src/aurora/ltrfile.cpp

```diff
@@ -0,0 +1,117 @@
+//
+// Created by patrick on 02.05.18.
+//
+
+#include "src/common/readstream.h"
+#include "src/common/strutil.h"
+
+#include "src/aurora/ltrfile.h"
+#include "src/aurora/resman.h"
+
+static const uint32 kLTRID  = MKTAG('L', 'T', 'R', ' ');
+static const uint32 kVersion10 = MKTAG('V', '1', '.', '0');
+
+static const char* kLetters = "abcdefghijklmnopqrstuvwxyz'-";
+
+namespace Aurora {
+
+LTRFile::LTRFile(Common::UString name) {
+	Common::ScopedPtr<Common::SeekableReadStream> stream(ResMan.getResource(name, kFileTypeLTR));
+
+	readHeader(*stream);
+
+	if (_id != kLTRID)
+		throw Common::Exception("Not a LTR file (%s)", Common::debugTag(_id).c_str());
+
+	if (_version != kVersion10)
+		throw Common::Exception("Unsupported LTR file version %s", Common::debugTag(_version).c_str());
+
+	byte letterCount = stream->readByte();
+	if (letterCount != 28)
+		throw Common::Exception("Unsupported Letter count %i", letterCount);
+
+	stream->read(&_singleLetters, sizeof(LetterSet));
```

Comment:
Don't do that, please. Due to struct alignment, endianness and potentially float format differences, this is not portable.

Instead, please read the individual floats in loops with readIEEEFloatLE().